crates/plugins/src/error.rs

```diff
@@ -0,0 +1,89 @@
+pub type PluginLookupResult<T> = std::result::Result<T, Error>;
+
+/// Error message during plugin lookup or deserializing
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("{0}")]
+    NotFound(NotFoundError),
+
+    #[error("{0}")]
+    ConnectionFailed(ConnectionFailedError),
+
+    #[error("{0}")]
+    InvalidManifest(InvalidManifestError),
+
+    #[error("URL parse error {0}")]
+    UrlParseError(url::ParseError),
```

Comment:
A convenience from `thiserror`:

```suggestion
    UrlParseError(#[from] url::ParseError),
```

This generates a `impl From<url::ParseError> for Error` for you, which allows `?` to convert types for you without an explicit `map_err`.